lib/install/management/upgrade.go

```diff
@@ -99,10 +101,37 @@ func (d *Dispatcher) Upgrade(vch *vm.VirtualMachine, conf *config.VirtualContain
 
 	d.deleteUpgradeImages(ds, settings)
 	d.retryDeleteSnapshot(snapshotName, conf.Name)
+
 	// return the error message for upgrade
 	return err
 }
 
+func (d *Dispatcher) Rollback(vch *vm.VirtualMachine, conf *config.VirtualContainerHostConfigSpec, settings *data.InstallerData) error {
+
+	// some setup that is only necessary because we didn't just create a VCH in this case
+	d.appliance = vch
+	d.setDockerPort(conf, settings)
+	if !strings.HasSuffix(conf.Target, "/sdk") {
+		conf.Target = fmt.Sprintf("%s/sdk", conf.Target)
+	}
+
+	snapshot, err := d.appliance.GetCurrentSnapshotTree(d.ctx)
+	if err != nil {
+		return errors.Errorf("could not find a snapshot to roll back to: %s", err)
```

Comment:
the relationship between the old upgrade version and snapshot might not be clear enough for customer. here might add some more meaningful information, sth like no upgrade version is found